File: LinkedLists/reverse/reverse.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None


def print_ll(ll):
    while ll != None:
        print(ll.data, end=" ")
        ll = ll.next

def reverse(cur):
    next = prev = None
    while cur != None:
        next = cur.next
        cur.next = prev
        prev = cur
        cur = next
    return prev


ll = Node(1)
ll.next = Node(2)
ll.next.next = Node(3)
ll.next.next.next = Node(4)
ll.next.next.next.next = Node(5)

print_ll(ll)
head = reverse(ll)
print("Head is ", head.data)
print("Reverse is:")
print_ll(head)

File: LinkedLists/reverse/reverse.js
class Node{
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}


function print_ll(node){
  let ll = node;
  while (ll != null){
    console.log(ll.data)
    ll = ll.next
  }
}
function reverse(head) {
  let cur = head
  let next = prev = null
  while (cur !== null){
    next = cur.next
    cur.next = prev
    prev = cur
    cur = next}
  return prev
}


const ll = new Node(1)
ll.next = new Node(2)
ll.next.next = new Node(3)
ll.next.next.next = new Node(4)
ll.next.next.next.next = new Node(5)

print_ll(ll)
head = reverse(ll)
console.log("Reverse is:")
print_ll(head)
